colour/recovery/jakob2019.py

```diff
@@ -680,49 +751,79 @@ def optimize(L, coefficients_0):
                     illuminant,
                     coefficients_0,
                     dimensionalise=False,
-                    use_feedback="adaptive-from-grey",
-                    try_directly_first=True)
+                    try_directly_first=True,
+                    use_feedback='adaptive-from-grey')
 
                 if verbose:
-                    print('%.4f %.4f %.4f  %13.6g %13.6g %13.6g  %g' %
-                          (RGB[0], RGB[1], RGB[2], coefficients[0],
-                           coefficients[1], coefficients[2], error))
+                    print_callable(
+                        '{0:.4f} {1:.4f} {2:.4f}  '
+                        '{3:>13.6f} {4:>13.6f} {5:>13.6f} {6:.6f}'.format(
+                            RGB[0], RGB[1], RGB[2], coefficients[0],
+                            coefficients[1], coefficients[2], error))
 
                 self.coefficients[i, L, j, k, :] = dimensionalise_coefficients(
                     coefficients, cmfs.shape)
 
                 return coefficients
 
-            # Start from somewhere in the middle, similarly to how feedback
+            # Starts from somewhere in the middle, similarly to how feedback
             # works in find_coefficients.
             middle_L = lightness_steps // 3
             middle_coefficients = optimize(middle_L, (0, 0, 0))
 
-            # Go down the lightness scale
+            # Goes down the lightness scale.
             coefficients_0 = middle_coefficients
             for L in reversed(range(0, middle_L)):
                 coefficients_0 = optimize(L, coefficients_0)
 
-            # Go up the lightness scale
+            # Goes up the lightness scale.
             coefficients_0 = middle_coefficients
             for L in range(middle_L + 1, lightness_steps):
                 coefficients_0 = optimize(L, coefficients_0)
 
-        self.res = lightness_steps
-        self.__setup_cubes()
+        self.resolution = lightness_steps
+        self._create_cube()
+
+    def read(self, path):
+        """
+        Loads a lookup table from a *.coeff* file.
+
+        Parameters
+        ----------
+        path : unicode
+            Path to the file.
+        """
+
+        with open(path, 'rb') as coeff_file:
+            if coeff_file.read(4).decode('ISO-8859-1') != 'SPEC':
+                raise ValueError(
+                    'Bad magic number, this is likely not the right file type!'
+                )
+
+            self.resolution = struct.unpack('i', coeff_file.read(4))[0]
+            self.scale = np.fromfile(
+                coeff_file, count=self.resolution, dtype=np.float32)
+            self.coefficients = np.fromfile(
+                coeff_file,
+                count=3 * self.resolution ** 3 * 3,
```

Comment:
Missing space, and would also re commend adding braces to clarify precedence.

Reply:
Updated too!